pennylane/ops/functions/evolve.py

```diff
@@ -14,18 +14,40 @@
 """
 This module contains the qml.evolve function.
 """
-from typing import Union
+from typing import Union, Tuple
 
 from pennylane.operation import Operator
 from pennylane.ops import Evolution
 from pennylane.pulse import ParametrizedEvolution, ParametrizedHamiltonian
+import pennylane as qml
 
 
-def evolve(op: Union[Operator, ParametrizedHamiltonian]):
-    """Returns a new operator that computes the evolution of ``op``.
+def evolve(
+    op: Union[Operator, ParametrizedHamiltonian],
+    t: Union[float, Tuple[float]] = None,
+    dt: float = None,
+):
+    r"""Returns a new operator that computes the evolution of ``op``.
 
     Args:
         op (Union[.Operator, .ParametrizedHamiltonian]): operator to evolve
+        t (Union[float, Tuple[float]]): Time.
+
+            * If ``op`` is a :class:`.Operator`, ``t`` corresponds to the coefficient multiplying
+                the exponentiated operator: :math:`\exp\{i\bm{O}t)\}`. If ``None``, ``t=1`` is used.
+
+            * If ``op`` is a :class:`.ParametrizedHamiltonian`, ``t`` can be either a float or a
+                tuple of floats. If a float, the operator is evolved from ``0`` to ``t``. If a tuple of
+                floats, each value corresponds to the start and end time of the evolution. Note that such
+                absolute times only have meaning within an instance of ``ParametrizedEvolution`` and
+                will not affect other gates.
+
+        dt (float): Time step used.
+
+            * If ``op`` is a :class:`.ParametrizedHamiltonian`, ``dt`` will be
+                used to convert the initial and final time values into a list of values that the ``odeint``
+                solver will use. The solver might perform intermediate steps if necessary. It is recommended
+                to not provide a value for ``dt``.
```

Comment:
This is missing the "if ``op`` is a :class:`~.Operator`, it will be used for a decomposition" bullet point.

Are we sure these are _really_ want both the `Operator` case and the `ParametrizedHamiltonian` case to be covered by the same  function if the arguments mean different things and take different forms depending on the type of the first argument?

Reply:
The missing text is added in [this PR](https://github.com/PennyLaneAI/pennylane/pull/3691).

Regarding your question, this is a UI matter, I'm sure @trbromley will have a better answer.

Reply:
We now have two different functions and use singledispatch to choose the correct function!